### ISSUE TITLE
a promise was created in a handler but was not returned from it,

### ISSUE THREAD

**Issue by soluty** [moleculerjs/moleculer#381]
when i try use moleculer-prometheus for do some metrics work,  but there is a warning like 
a promise was created in a handler but was not returned from it 。
i found if someone if listen $services.changed events , it will get this warning.  i have no idea how to fix this now

**Comment by soluty** [comment]
i read the emitLocalServices function in event-catalog.js file. it seems like wether or not the event handler function is async , all the handlers will exce  (parallels). they dont need to await previous handler to exec done. is it right?

**Comment by soluty** [comment]
i found the real place lost  Promise return.  in service.js file  _start function code like this:
```
  _start() {
    return this.Promise
      .resolve()
      .then(() => {
        return this.broker.middlewares.callHandlers('serviceStarting', [this])
      })
      .then(() => {
        // Wait for dependent services
        if (this.schema.dependencies) return this.waitForServices(this.schema.dependencies, this.settings.$dependencyTimeout || 0)
      })
      .then(() => {
        if (_.isFunction(this.schema.started)) return this.Promise.method(this.schema.started).call(this)

        if (Array.isArray(this.schema.started)) {
          return this.schema.started.map(fn => this.Promise.method(fn.bind(this))).reduce((p, fn) => p.then(fn), this.Promise.resolve())
        }
      })
      .then(() => {
        // Register service
        this.broker.registerLocalService(this._serviceSpecification)
        // this line lost return , add return null then no warning 
        return null
      })
      .then(() => {
        return this.broker.middlewares.callHandlers('serviceStarted', [this])
      })
  }
```
can you fix this?

**Comment by icebob** [comment]
Fixed. Will be released in 0.13.3
